doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1311 @@
+=pod
+
+=head1 NAME
+
+migration_guide - OpenSSL migration guide.
+
+=head1 SYNOPSIS
+
+See the individual manual pages for details.
+
+=head1 DESCRIPTION
+
+This guide details the changes and how to change your code to migrate to new
+versions of OpenSSL. Currently this covers OpenSSL 3.0. For earlier versions
+refer to L<https://github.com/openssl/openssl/blob/master/CHANGES.md>
+
+=head2 Migrating to OpenSSL 3.0
+
+Most existing applications should just need to be recompiled, with
+little to no changes.
+The majority of the changes in OpenSSL 3.0 are internal architectural
+restructuring changes to facilitate a long-term supportable cryptographic
+framework. This enables better separation of algorithm implementations from
+algorithm APIs. These structural changes also support a more maintainable
+OpenSSL FIPS Cryptographic Module.
+Many low-level functions have been marked as deprecated APIs, but no previously
```

Comment:
Not entirely true. `EVP_PKEY_set_alias_type` was removed entirely.

Reply:
Yes - might be best to omit that altogether..